# **DATA LOADING AND EXPLORATION**:

In [118]:
#importing libraries 
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.metrics import accuracy_score, classification_report
import pandas as pd
from sklearn.metrics import accuracy_score
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
import folium

In [119]:
#loading  and previewing rainfall data sets:
rainfal=pd.read_csv("kenya_rainfall_2022_2024.csv")
rainfal.head()

time  precipitation_sum (mm)  Unnamed: 2  Unnamed: 3  Unnamed: 4  \
0  01/01/2022                     0.0         NaN         NaN         NaN   
1  02/01/2022                     0.0         NaN         NaN         NaN   
2  03/01/2022                     0.0         NaN         NaN         NaN   
3  04/01/2022                     0.0         NaN         NaN         NaN   
4  05/01/2022                     0.0         NaN         NaN         NaN   

   Unnamed: 5     town  
0         NaN  Nairobi  
1         NaN  Nairobi  
2         NaN  Nairobi  
3         NaN  Nairobi  
4         NaN  Nairobi

In [120]:
#loading and previewing wfp food prices
foodprices=pd.read_csv('wfp_food_prices_ken.csv')
foodprices.head()

date      admin1      admin2            market         market_id  \
0       #date  #adm1+name  #adm2+name  #loc+market+name  #loc+market+code   
1  2006-01-15       Coast     Mombasa           Mombasa               191   
2  2006-01-15     Eastern       Kitui             Kitui               187   
3  2006-01-15     Eastern       Kitui             Kitui               187   
4  2006-01-15     Eastern       Kitui             Kitui               187   

   latitude longitude            category         commodity commodity_id  \
0  #geo+lat  #geo+lon          #item+type        #item+name   #item+code   
1     -4.05     39.67     pulses and nuts             Beans           50   
2     -1.37     38.02  cereals and tubers  Potatoes (Irish)          148   
3     -1.37     38.02  cereals and tubers           Sorghum           65   
4     -1.37     38.02     pulses and nuts       Beans (dry)          262   

         unit         priceflag         pricetype        currency    price  \
0  #item+unit  #item+price+flag  #item+price+type  #currency+code   #value   
1          KG            actual         Wholesale             KES    33.63   
2       50 KG            actual         Wholesale             KES  1249.99   
3       90 KG            actual         Wholesale             KES     1800   
4          KG            actual            Retail             KES       39   

     usdprice  
0  #value+usd  
1        0.47  
2       17.39  
3       25.03  
4        0.54

In [121]:
#loading and previewing cities data:
towns=pd.read_csv('cities_latitude_and_longitudes.csv')
towns.head()

city     lat      lng country iso2    admin_name  capital  population  \
0  Nairobi -1.2864  36.8172   Kenya   KE  Nairobi City  primary   5545000.0   
1  Mombasa -4.0500  39.6667   Kenya   KE       Mombasa    admin   1208333.0   
2   Nakuru -0.3000  36.0667   Kenya   KE        Nakuru    admin    570674.0   
3  Eldoret  0.5167  35.2833   Kenya   KE   Uasin Gishu    admin    475716.0   
4   Kisumu -0.0833  34.7667   Kenya   KE        Kisumu    admin    409928.0   

   population_proper  
0          5545000.0  
1          1200000.0  
2           307990.0  
3           289380.0  
4           409928.0

In [122]:
#previewing foodprices columns
list(foodprices.columns)

['date',
 'admin1',
 'admin2',
 'market',
 'market_id',
 'latitude',
 'longitude',
 'category',
 'commodity',
 'commodity_id',
 'unit',
 'priceflag',
 'pricetype',
 'currency',
 'price',
 'usdprice']

In [123]:
#previewing rainfal columns
list(rainfal.columns)

['time',
 'precipitation_sum (mm)',
 'Unnamed: 2',
 'Unnamed: 3',
 'Unnamed: 4',
 'Unnamed: 5',
 'town']

In [124]:
foodprices.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12783 entries, 0 to 12782
Data columns (total 16 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   date          12783 non-null  object
 1   admin1        12749 non-null  object
 2   admin2        12749 non-null  object
 3   market        12783 non-null  object
 4   market_id     12783 non-null  object
 5   latitude      12749 non-null  object
 6   longitude     12749 non-null  object
 7   category      12783 non-null  object
 8   commodity     12783 non-null  object
 9   commodity_id  12783 non-null  object
 10  unit          12783 non-null  object
 11  priceflag     12783 non-null  object
 12  pricetype     12783 non-null  object
 13  currency      12783 non-null  object
 14  price         12783 non-null  object
 15  usdprice      12783 non-null  object
dtypes: object(16)
memory usage: 1.6+ MB


Food price data has 12783 rows.
columns such as date price are of string data types.

In [125]:
#Checking for null values:
foodprices.isnull().sum()

date             0
admin1          34
admin2          34
market           0
market_id        0
latitude        34
longitude       34
category         0
commodity        0
commodity_id     0
unit             0
priceflag        0
pricetype        0
currency         0
price            0
usdprice         0
dtype: int64

Food prices has a total of 34 rows with null values under the region, subregion,latitude and longitude columns.

In [126]:
rainfal.isnull().sum()

time                         61
precipitation_sum (mm)       61
Unnamed: 2                69109
Unnamed: 3                69109
Unnamed: 4                69109
Unnamed: 5                69109
town                         61
dtype: int64

Unnamed columns in rainfall dataset have 69109 null values while time and precipitation columsn have 61 missing values

In [127]:
rainfal.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 69109 entries, 0 to 69108
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   time                    69048 non-null  object 
 1   precipitation_sum (mm)  69048 non-null  float64
 2   Unnamed: 2              0 non-null      float64
 3   Unnamed: 3              0 non-null      float64
 4   Unnamed: 4              0 non-null      float64
 5   Unnamed: 5              0 non-null      float64
 6   town                    69048 non-null  object 
dtypes: float64(5), object(2)
memory usage: 3.7+ MB


Rainfal data has 69048 rows. Time column has string data type.
The data has 4 columns with null values 

In [128]:
#loading and previewing crop_yeilds
crop_yeilds=pd.read_csv('hvstat_africa_data_v1.0.csv')
crop_yeilds.head()

fnid country country_code admin_1 admin_2 product season_name  \
0  AO2008A101  Angola           AO   Bengo    none  Banana        Main   
1  AO2008A101  Angola           AO   Bengo    none  Banana        Main   
2  AO2008A101  Angola           AO   Bengo    none  Banana        Main   
3  AO2008A101  Angola           AO   Bengo    none  Banana        Main   
4  AO2008A101  Angola           AO   Bengo    none  Banana        Main   

   planting_year  planting_month  harvest_year  harvest_month  \
0           2007              10          2008              4   
1           2008              10          2009              4   
2           2014              10          2015              4   
3           2015              10          2016              4   
4           2016              10          2017              4   

  crop_production_system  qc_flag    area  production      yield  
0               All (PS)        0  1364.0     19588.0  14.360704  
1               All (PS)        0  1364.0     20461.0  15.000733  
2               All (PS)        0  8706.0    222044.0  25.504709  
3               All (PS)        0  8706.0    222044.0  25.504709  
4               All (PS)        0  9549.7    233967.0  24.499932

In [129]:
#extracting kenyan food_yeild data:
Kenya_crop_yeilds=crop_yeilds[crop_yeilds['country'].str.lower()=='kenya']
Kenya_crop_yeilds.head()

fnid country country_code  admin_1 admin_2        product  \
79170  KE2013A101   Kenya           KE  Mombasa    none         Banana   
79171  KE2013A101   Kenya           KE  Mombasa    none         Banana   
79172  KE2013A101   Kenya           KE  Mombasa    none         Banana   
79173  KE2013A101   Kenya           KE  Mombasa    none  Beans (mixed)   
79174  KE2013A101   Kenya           KE  Mombasa    none  Beans (mixed)   

      season_name  planting_year  planting_month  harvest_year  harvest_month  \
79170      Annual           2013               1          2012             12   
79171      Annual           2014               1          2013             12   
79172      Annual           2023               1          2022             12   
79173      Annual           1986               1          1985             12   
79174      Annual           1987               1          1986             12   

      crop_production_system  qc_flag   area  production      yield  
79170               All (PS)        0  125.1      1110.0   8.872902  
79171               All (PS)        0  121.5       952.0   7.835391  
79172               All (PS)        0   39.0       600.0  15.384615  
79173               All (PS)        0    3.0         2.0   0.666667  
79174               All (PS)        0    7.0         6.0   0.857143

#  **DATA PRE PROCESSING AND FEATURE ENGINEERING**:


In [130]:
#renaming columns in food price data:
foodprices=foodprices.rename(columns={"date": "Date",
    "admin1": "Region",
    "admin2": "Subregion",
    "market": "Market",
    "market_id": "Market_ID",
    "latitude": "Latitude",
    "longitude": "Longitude",
    "category": "Category",
    "commodity": "Commodity",
    "commodity_id": "Commodity_ID",
    "unit": "Unit",
    "priceflag": "Price_Flag",
    "pricetype": "Price_Type",
    "currency": "Currency",
    "price": "Price_KES",
    "usdprice": "Price_USD"  
})
foodprices.head()

Date      Region   Subregion            Market         Market_ID  \
0       #date  #adm1+name  #adm2+name  #loc+market+name  #loc+market+code   
1  2006-01-15       Coast     Mombasa           Mombasa               191   
2  2006-01-15     Eastern       Kitui             Kitui               187   
3  2006-01-15     Eastern       Kitui             Kitui               187   
4  2006-01-15     Eastern       Kitui             Kitui               187   

   Latitude Longitude            Category         Commodity Commodity_ID  \
0  #geo+lat  #geo+lon          #item+type        #item+name   #item+code   
1     -4.05     39.67     pulses and nuts             Beans           50   
2     -1.37     38.02  cereals and tubers  Potatoes (Irish)          148   
3     -1.37     38.02  cereals and tubers           Sorghum           65   
4     -1.37     38.02     pulses and nuts       Beans (dry)          262   

         Unit        Price_Flag        Price_Type        Currency Price_KES  \
0  #item+unit  #item+price+flag  #item+price+type  #currency+code    #value   
1          KG            actual         Wholesale             KES     33.63   
2       50 KG            actual         Wholesale             KES   1249.99   
3       90 KG            actual         Wholesale             KES      1800   
4          KG            actual            Retail             KES        39   

    Price_USD  
0  #value+usd  
1        0.47  
2       17.39  
3       25.03  
4        0.54

In [131]:
# converting foodprices prices to float:
foodprices['Price_USD']=pd.to_numeric(foodprices['Price_USD'],errors='coerce')
foodprices['Price_KES']=pd.to_numeric(foodprices['Price_KES'],errors='coerce')
#changeing date to date format
foodprices['Date']=pd.to_datetime(foodprices['Date'],format='%Y-%m-%d',errors='coerce')
foodprices.head(2)

Date      Region   Subregion            Market         Market_ID  \
0        NaT  #adm1+name  #adm2+name  #loc+market+name  #loc+market+code   
1 2006-01-15       Coast     Mombasa           Mombasa               191   

   Latitude Longitude         Category   Commodity Commodity_ID        Unit  \
0  #geo+lat  #geo+lon       #item+type  #item+name   #item+code  #item+unit   
1     -4.05     39.67  pulses and nuts       Beans           50          KG   

         Price_Flag        Price_Type        Currency  Price_KES  Price_USD  
0  #item+price+flag  #item+price+type  #currency+code        NaN        NaN  
1            actual         Wholesale             KES      33.63       0.47

In [132]:
#removing rows in food price with misisng criticla values:
foodprices.dropna(subset=['Commodity','Price_KES','Date','Market'],inplace=True)
foodprices.head(2)

Date   Region Subregion   Market Market_ID Latitude Longitude  \
1 2006-01-15    Coast   Mombasa  Mombasa       191    -4.05     39.67   
2 2006-01-15  Eastern     Kitui    Kitui       187    -1.37     38.02   

             Category         Commodity Commodity_ID   Unit Price_Flag  \
1     pulses and nuts             Beans           50     KG     actual   
2  cereals and tubers  Potatoes (Irish)          148  50 KG     actual   

  Price_Type Currency  Price_KES  Price_USD  
1  Wholesale      KES      33.63       0.47  
2  Wholesale      KES    1249.99      17.39

In [133]:
foodprices.info()

<class 'pandas.core.frame.DataFrame'>
Index: 12782 entries, 1 to 12782
Data columns (total 16 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   Date          12782 non-null  datetime64[ns]
 1   Region        12748 non-null  object        
 2   Subregion     12748 non-null  object        
 3   Market        12782 non-null  object        
 4   Market_ID     12782 non-null  object        
 5   Latitude      12748 non-null  object        
 6   Longitude     12748 non-null  object        
 7   Category      12782 non-null  object        
 8   Commodity     12782 non-null  object        
 9   Commodity_ID  12782 non-null  object        
 10  Unit          12782 non-null  object        
 11  Price_Flag    12782 non-null  object        
 12  Price_Type    12782 non-null  object        
 13  Currency      12782 non-null  object        
 14  Price_KES     12782 non-null  float64       
 15  Price_USD     12782 non-null  float64    

In [134]:
foodprices.describe()

Date     Price_KES     Price_USD
count                          12782  12782.000000  12782.000000
mean   2019-07-26 00:59:35.778438400   1780.668971     16.436858
min              2006-01-15 00:00:00      5.000000      0.044000
25%              2017-02-15 00:00:00     60.000000      0.570000
50%              2021-06-15 00:00:00    150.000000      1.210000
75%              2023-01-15 00:00:00   2644.000000     25.237500
max              2025-05-15 00:00:00  19800.000000    184.820000
std                              NaN   2905.224440     25.793262

In [135]:
#checking for distinct commodity names
commodity_names=foodprices['Commodity'].value_counts()
commodity_names

Commodity
Beans (dry)                  1155
Maize (white)                1153
Maize                        1009
Potatoes (Irish)              846
Sorghum                       646
Beans                         547
Oil (vegetable)               527
Wheat flour                   422
Salt                          412
Rice                          408
Sugar                         383
Maize flour                   376
Tomatoes                      346
Milk (UHT)                    339
Cabbage                       319
Bananas                       318
Kale                          317
Maize (white, dry)            278
Onions (dry)                  273
Beans (yellow)                246
Potatoes (Irish, white)       238
Meat (beef)                   208
Meat (goat)                   202
Cowpeas                       174
Meat (camel)                  160
Beans (rosecoco)              160
Rice (aromatic)               149
Beans (dolichos)              138
Bread                         126
Milk

In [136]:
foodpricesmarkes=foodprices['Market'].value_counts()
foodpricesmarkes.head()

Market
Nairobi                       1231
Eldoret town (Uasin Gishu)     957
Kisumu                         673
Mombasa                        448
Kitui                          440
Name: count, dtype: int64

In [137]:
#standardizing text formats:
foodprices['Commodity']=foodprices['Commodity'].str.strip().str.lower()
foodprices['Category']=foodprices['Category'].str.strip().str.lower()
foodprices['Region']=foodprices['Region'].str.strip().str.lower()

In [138]:
#Standardizing commodity naming:
commodity_map={ "beans (dry)": "dry beans",
    "beans (mung)": "mung beans",
    "beans (dolichos)": "dolichos beans","maize (white)":"fresh white maize",
     "maize": "dry white maize",
    "maize (white, dry)": "dry white maize",
    "potatoes (irish)": "irish potatoes",
    "potatoes (irish, red)": "red irish potatoes",
    "potatoes (irish, white)": "white irish potatoes",
    "milk (cow, pasteurized)": "pasteurized cow milk",
    "milk (cow, fresh)": "fresh cow milk",
    "milk (camel, fresh)": "fresh camel milk",
    "milk (uht)": "UHT milk",
    "meat (camel)": "camel meat",
    "meat (goat)": "goat meat",
    "meat (beef)": "beef",
    "fish (omena, dry)": "dry omena fish",
    "fuel (diesel)": "diesel",
    "fuel (kerosene)": "kerosene",
    "fuel (petrol-gasoline)": "petrol",
    "oil (vegetable)": "vegetable oil",
    "wheat flour": "wheat flour",
    "maize flour": "maize flour",
    "onions (dry)": "dry onions",
    "onions (red)": "red onions",
    "cowpea leaves": "cowpea leaves",
    "cooking fat": "cooking fat",
    'red onions':'dry onions'
               
              }

In [139]:
foodprices['Commodity']=foodprices['Commodity'].replace(commodity_map)

In [140]:
commodity_names=foodprices['Commodity'].value_counts()
commodity_names

Commodity
dry white maize              1287
dry beans                    1155
fresh white maize            1153
irish potatoes                846
sorghum                       646
beans                         547
vegetable oil                 527
wheat flour                   422
salt                          412
rice                          408
sugar                         383
maize flour                   376
tomatoes                      346
UHT milk                      339
cabbage                       319
bananas                       318
kale                          317
dry onions                    273
beans (yellow)                246
white irish potatoes          238
beef                          208
goat meat                     202
cowpeas                       174
camel meat                    160
beans (rosecoco)              160
rice (aromatic)               149
dolichos beans                138
bread                         126
pasteurized cow milk          118
red 

In [141]:
#unique commodity per market:
Commodity_counts=foodprices.groupby ('Market')['Commodity'].nunique().reset_index(name='Unique_commodities')
Commodity_counts=Commodity_counts.sort_values(by='Unique_commodities',ascending=False)
Commodity_counts .head()

Market  Unique_commodities
44      Marigat town (Baringo)                  34
24           Kangemi (Nairobi)                  34
7       Garissa town (Garissa)                  30
4   Eldoret town (Uasin Gishu)                  24
33         Kitengela (Kajiado)                  21

In [142]:
foodprices.head()

Date   Region Subregion    Market Market_ID Latitude Longitude  \
1 2006-01-15    coast   Mombasa   Mombasa       191    -4.05     39.67   
2 2006-01-15  eastern     Kitui     Kitui       187    -1.37     38.02   
3 2006-01-15  eastern     Kitui     Kitui       187    -1.37     38.02   
4 2006-01-15  eastern     Kitui     Kitui       187    -1.37     38.02   
5 2006-01-15  eastern  Marsabit  Marsabit       190     2.33     37.98   

             Category          Commodity Commodity_ID   Unit Price_Flag  \
1     pulses and nuts              beans           50     KG     actual   
2  cereals and tubers     irish potatoes          148  50 KG     actual   
3  cereals and tubers            sorghum           65  90 KG     actual   
4     pulses and nuts          dry beans          262     KG     actual   
5  cereals and tubers  fresh white maize           67     KG     actual   

  Price_Type Currency  Price_KES  Price_USD  
1  Wholesale      KES      33.63       0.47  
2  Wholesale      KES    1249.99      17.39  
3  Wholesale      KES    1800.00      25.03  
4     Retail      KES      39.00       0.54  
5     Retail      KES      21.00       0.29

In [143]:
#checking unique markets using Market column
unique_markets=foodprices['Market'].unique()
print (unique_markets)

['Mombasa' 'Kitui' 'Marsabit' 'Nairobi' 'Mandera' 'Kisumu'
 'Lodwar (Turkana)' 'Eldoret town (Uasin Gishu)' 'Nakuru'
 'Hola (Tana River)' 'Kilifi' 'Garissa' 'Marigat (Baringo)' 'Kajiado'
 'Karatina (Nyeri)' 'Vanga (Kwale)' 'Kitui town (Kitui)' 'Makueni'
 'Wote town (Makueni)' 'Kitengela (Kajiado)' 'Garissa town (Garissa)'
 'Takaba (Mandera)' 'Marigat town (Baringo)'
 'Illbissil Food Market (Kajiado)' 'Wakulima (Nakuru)'
 'Kongowea (Mombasa)' 'Tala Centre Market (Machakos)' 'Kangemi (Nairobi)'
 'Kibuye (Kisumu)' 'Makutano (West Pokot)' 'Kathonzweni (Makueni)'
 'Kaanwa (Tharaka Nithi)' 'Lomut (West Pokot)' 'Wakulima (Nairobi)'
 'Dagahaley (Daadab)' 'Bangladesh (Mombasa)' 'Kalahari (Mombasa)'
 'Shonda (Mombasa)' 'Kawangware (Nairobi)' 'Kibra (Nairobi)'
 'Mathare (Nairobi)' 'Mukuru (Nairobi)' 'Hagadera (Daadab)'
 'Kalobeyei (Village 1)' 'Kalobeyei (Village 2)' 'Kalobeyei (Village 3)'
 'Isiolo town' 'Dandora (Nairobi)' 'Wajir town' 'Ethiopia (Kakuma)'
 'HongKong (Kakuma)' 'Kakuma 2' 'Kakuma

In [144]:
#Extracting common market name from Market column:
def extract_first_name(market):
    if '(' in market:
        market=market.split('(')[0]
    return market.split('(')[0]
foodprices['Market_Name']=foodprices['Market'].apply(extract_first_name)

In [145]:
#To check unique Market_Name
Unique_foodprices_Market_Name=foodprices['Market_Name'].unique()
print (Unique_foodprices_Market_Name)

['Mombasa' 'Kitui' 'Marsabit' 'Nairobi' 'Mandera' 'Kisumu' 'Lodwar '
 'Eldoret town ' 'Nakuru' 'Hola ' 'Kilifi' 'Garissa' 'Marigat ' 'Kajiado'
 'Karatina ' 'Vanga ' 'Kitui town ' 'Makueni' 'Wote town ' 'Kitengela '
 'Garissa town ' 'Takaba ' 'Marigat town ' 'Illbissil Food Market '
 'Wakulima ' 'Kongowea ' 'Tala Centre Market ' 'Kangemi ' 'Kibuye '
 'Makutano ' 'Kathonzweni ' 'Kaanwa ' 'Lomut ' 'Dagahaley ' 'Bangladesh '
 'Kalahari ' 'Shonda ' 'Kawangware ' 'Kibra ' 'Mathare ' 'Mukuru '
 'Hagadera ' 'Kalobeyei ' 'Isiolo town' 'Dandora ' 'Wajir town'
 'Ethiopia ' 'HongKong ' 'Kakuma 2' 'Kakuma 3' 'Kakuma 4' 'Lodwar town'
 'Mogadishu ' 'Junda ' 'Moroto ' 'Dadaab town' 'IFO ' 'Kisumu Ndogo '
 'Marsabit town']


In [146]:
#cleaning market_name to drop unnecssary words in the name
foodprices['Market_Name']=(foodprices['Market_Name'].str.replace(r'\b(town|food|centre)\b','',case=False,regex=True).str.replace(r'|s+', '',regex=True).str.strip()
                          .str.title())

#setting  title case:
foodprices['Region']=foodprices['Region'].str.title()
foodprices['Subregion']=foodprices['Subregion'].str.title()
foodprices['Category']=foodprices['Category'].str.title()
foodprices['Category']=foodprices['Category'].str.title() 
foodprices['Commodity']=foodprices['Commodity'].str.title() 

In [147]:
Unique_foodprices_Market_Name=foodprices['Market_Name'].unique()
print (Unique_foodprices_Market_Name)

['Mombaa' 'Kitui' 'Marabit' 'Nairobi' 'Mandera' 'Kiumu' 'Lodwar' 'Eldoret'
 'Nakuru' 'Hola' 'Kilifi' 'Garia' 'Marigat' 'Kajiado' 'Karatina' 'Vanga'
 'Makueni' 'Wote' 'Kitengela' 'Takaba' 'Illbiil  Market' 'Wakulima'
 'Kongowea' 'Tala  Market' 'Kangemi' 'Kibuye' 'Makutano' 'Kathonzweni'
 'Kaanwa' 'Lomut' 'Dagahaley' 'Bangladeh' 'Kalahari' 'Shonda' 'Kawangware'
 'Kibra' 'Mathare' 'Mukuru' 'Hagadera' 'Kalobeyei' 'Iiolo' 'Dandora'
 'Wajir' 'Ethiopia' 'Hongkong' 'Kakuma 2' 'Kakuma 3' 'Kakuma 4' 'Mogadihu'
 'Junda' 'Moroto' 'Dadaab' 'Ifo' 'Kiumu Ndogo']


In [148]:
unique_Subregion=foodprices['Subregion'].unique()
print (unique_Subregion)

['Mombasa' 'Kitui' 'Marsabit' 'Nairobi' 'Mandera' 'Kisumu' 'Turkana'
 'Uasin Gishu' 'Nakuru' nan 'Kilifi' 'Garissa' 'Baringo' 'Kajiado' 'Nyeri'
 'Kwale' 'Makueni' 'Machakos' 'West Pokot' 'Meru South' 'Isiolo' 'Wajir']


In [149]:
cleaned_food_prices=foodprices
cleaned_food_prices.head()


Date   Region Subregion    Market Market_ID Latitude Longitude  \
1 2006-01-15    Coast   Mombasa   Mombasa       191    -4.05     39.67   
2 2006-01-15  Eastern     Kitui     Kitui       187    -1.37     38.02   
3 2006-01-15  Eastern     Kitui     Kitui       187    -1.37     38.02   
4 2006-01-15  Eastern     Kitui     Kitui       187    -1.37     38.02   
5 2006-01-15  Eastern  Marsabit  Marsabit       190     2.33     37.98   

             Category          Commodity Commodity_ID   Unit Price_Flag  \
1     Pulses And Nuts              Beans           50     KG     actual   
2  Cereals And Tubers     Irish Potatoes          148  50 KG     actual   
3  Cereals And Tubers            Sorghum           65  90 KG     actual   
4     Pulses And Nuts          Dry Beans          262     KG     actual   
5  Cereals And Tubers  Fresh White Maize           67     KG     actual   

  Price_Type Currency  Price_KES  Price_USD Market_Name  
1  Wholesale      KES      33.63       0.47      Mombaa  
2  Wholesale      KES    1249.99      17.39       Kitui  
3  Wholesale      KES    1800.00      25.03       Kitui  
4     Retail      KES      39.00       0.54       Kitui  
5     Retail      KES      21.00       0.29     Marabit

**DATA CLEANING RAINFALL DATA:**

In [150]:
#dropping unecessary columns in rainfall data
rainfal=rainfal.drop(columns=['Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5'])
rainfal.head()

time  precipitation_sum (mm)     town
0  01/01/2022                     0.0  Nairobi
1  02/01/2022                     0.0  Nairobi
2  03/01/2022                     0.0  Nairobi
3  04/01/2022                     0.0  Nairobi
4  05/01/2022                     0.0  Nairobi

In [151]:
#renaming columns innrainfal data 
rainfal=rainfal.rename(columns={'time':'Date','precipitation_sum (mm)':'Rainfall_MM',"town": "Town"})
rainfal.head()

Date  Rainfall_MM     Town
0  01/01/2022          0.0  Nairobi
1  02/01/2022          0.0  Nairobi
2  03/01/2022          0.0  Nairobi
3  04/01/2022          0.0  Nairobi
4  05/01/2022          0.0  Nairobi

In [152]:
#setting  title case:
rainfal['Town']=rainfal['Town'].str.title()

In [153]:
#converting  to date format and summarizing rainfall per month year
rainfal['Date']=pd.to_datetime(rainfal['Date'],dayfirst=True,errors='coerce')
rainfal['YearMonth']=rainfal['Date'].dt.to_period('M')
Mothly_Rainfal=rainfal.groupby(['Town','YearMonth'])['Rainfall_MM'].sum().reset_index()
Mothly_Rainfal=Mothly_Rainfal.sort_values(['Town','YearMonth'])
Mothly_Rainfal.head()

Town YearMonth  Rainfall_MM
0  Bungoma   2022-01         50.6
1  Bungoma   2022-02         37.3
2  Bungoma   2022-03         31.0
3  Bungoma   2022-04        218.4
4  Bungoma   2022-05        244.6

In [154]:
Mothly_Rainfal['Year']=Mothly_Rainfal['YearMonth'].dt.year
Mothly_Rainfal['Month']=Mothly_Rainfal['YearMonth'].dt.month
Mothly_Rainfal.head()

Town YearMonth  Rainfall_MM  Year  Month
0  Bungoma   2022-01         50.6  2022      1
1  Bungoma   2022-02         37.3  2022      2
2  Bungoma   2022-03         31.0  2022      3
3  Bungoma   2022-04        218.4  2022      4
4  Bungoma   2022-05        244.6  2022      5